packages/core/src/manifest.test.ts

```diff
@@ -94,26 +95,293 @@ test.serial('rename manifest', async t => {
+
+  // check that the contents were NOT persisted to original manifest
+  const orig = await new Manifest(forkedId).read();
+  t.is(orig.proxies.length, 1);
+  t.is(orig.proxies[0].address, '0x123');
+
+  // check that the contents were persisted to dev copy of manifest
+  const dev = await new Manifest(devId, devInstanceMetadata, os.tmpdir()).read();
+  t.is(dev.proxies.length, 2);
+  t.is(dev.proxies[0].address, '0x123');
+  t.is(dev.proxies[1].address, '0x456');
+
+  await deleteManifests(t, forkedId);
+  await deleteFile(t, `${os.tmpdir()}/openzeppelin-upgrades/hardhat-${devId}-${instanceId}.json`);
+});
+
+test.serial('forked chain from known network with actual name', async t => {
+  const forkedId = 80001;
+  const devId = 55555;
+
+  await deleteManifests(t, forkedId);
+
+  await writeTestManifest(`.openzeppelin/polygon-mumbai.json`);
+
+  const instanceId = '0xbbb';
+  const devInstanceMetadata = { networkName: 'hardhat', instanceId: instanceId, forkedNetwork: { chainId: forkedId } };
+
+  const manifest = new Manifest(devId, devInstanceMetadata, os.tmpdir());
+
+  await assertNewName(t, forkedId);
+  await manifest.lockedRun(async () => {
+    await fs.access(`${os.tmpdir()}/openzeppelin-upgrades/chain-${devId}-${instanceId}.lock`);
+    const data = await manifest.read();
+    data.proxies.push({
+      address: '0x456',
+      txHash: '0x0',
+      kind: 'uups',
+    });
+
+    await assertNewName(t, forkedId);
+    await manifest.write(data);
+    await assertNewName(t, forkedId); // original network file should not be changed
+  });
+  t.throwsAsync(fs.access(`.openzeppelin/chain-${forkedId}.lock`));
+  t.throwsAsync(fs.access(`${os.tmpdir()}/openzeppelin-upgrades/chain-${devId}-${instanceId}.lock`));
+  await fs.access(`${os.tmpdir()}/openzeppelin-upgrades/hardhat-${devId}-${instanceId}.json`);
+
+  // check that the contents were NOT persisted to original manifest
+  const orig = await new Manifest(forkedId).read();
+  t.is(orig.proxies.length, 1);
+  t.is(orig.proxies[0].address, '0x123');
+
+  // check that the contents were persisted to dev copy of manifest
+  const dev = await new Manifest(devId, devInstanceMetadata, os.tmpdir()).read();
+  t.is(dev.proxies.length, 2);
+  t.is(dev.proxies[0].address, '0x123');
+  t.is(dev.proxies[1].address, '0x456');
+
+  await deleteManifests(t, forkedId);
+  await deleteFile(t, `${os.tmpdir()}/openzeppelin-upgrades/hardhat-${devId}-${instanceId}.json`);
+});
+
+test.serial('forked chain, real manifest already locked', async t => {
+  const forkedId = 80001;
+  const devId = 55555;
+
+  const instanceId = '0xccc';
+  const devInstanceMetadata = { networkName: 'hardhat', instanceId: instanceId, forkedNetwork: { chainId: forkedId } };
+
+  const realManifest = new Manifest(forkedId);
+  const devManifest = new Manifest(devId, devInstanceMetadata, os.tmpdir());
+
+  try {
+    await realManifest.lockedRun(async () => {
+      await devManifest.read();
+    });
```

Comment:
@frangio This test increases the overall `core` test time from 6 seconds to 11 seconds since it needs to wait for the lock timeout. Wondering if you have any alternative suggestions.

Reply:
We can add a way to customize the lockfile parameters used by `Manifest`. If we pass `retries: 0` it shouldn't add extra delay I think.